README.md

```diff
@@ -1,2 +1,151 @@
+
+```
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+using namespace std;
+using namespace mlboard;
+int main()
+{
+    GOOGLE_PROTOBUF_VERIFY_VERSION;
+    // creating a filewriter object that is repnsible for logging the summary.
+    fileWriter f1("temp");
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    
+    // Creating a scaler summary
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",1,1.1,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",2,1.2,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",3,1.4,f1);
+    mlboard::SummaryWriter<mlboard::fileWriter>::scalar("tag",5,1.5,f1);
+    end = std::chrono::system_clock::now(); 
+  
+    std::chrono::duration<double> elapsed_seconds = end - start; 
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+  
+    std::cout << "finished computation at " << std::ctime(&end_time) 
+              << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+
+    // This will allow you to indicate that you have logged all your data.
+    f1.close();
+    google::protobuf::ShutdownProtobufLibrary();
+}
+```
+
+You can then use tensorboard to visualize the scaler by using a simple command: `tensorboard --logdir .`
+
+The above snippet of code would print the time taken for the scaler to log
+
+```
+elapsed time: 40.0163s
+```
+
+Alternatively you can execute the logging in an async manner which would be faster at many instances by using `std::async`, here is a snippet which allows you to do the same operation but much faster as compared to the previous code
+
+```
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+using namespace std;
+using namespace mlboard;
+int main()
+{
+    GOOGLE_PROTOBUF_VERIFY_VERSION;
+    fileWriter f1("temp");
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    
+    std::future<void> result1 = async(std::launch::async,  SummaryWriter<fileWriter>::scalar,"tag",1,1.1,std::ref(f1));
+    std::future<void> result2 = async(std::launch::async,  SummaryWriter<fileWriter>::scalar, "tag",2,1.2,std::ref(f1));
+    std::future<void> result3 = async(std::launch::async,  SummaryWriter<fileWriter>::scalar,"tag",3,1.3,std::ref(f1));
+    std::future<void> result4 = async(std::launch::async, SummaryWriter<fileWriter>::scalar, "tag",4,1.4,std::ref(f1));
+    result1.get();
+    result2.get();
+    result3.get();
+    result4.get();
+    end = std::chrono::system_clock::now(); 
+  
+    std::chrono::duration<double> elapsed_seconds = end - start; 
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+  
+    std::cout << "finished computation at " << std::ctime(&end_time) 
```

Comment:
Same about this I guess. I think we should calculate the final time after `f1.close()`

Reply:
Currently it wouldn't make a much difference since , there are only 4 data points which is less than default maximum queue size

Reply:
Yes, you're correct. It's in readme at this moment. I was using the code as a test.  But I thought it's good to change it.
Actually when I was using before `f1.close()` and keeping `flushsec =2000` (2secs) I was getting time elapsed `0.0000587 ms`. And the program continues to  run. But it is wrong, as the thread is waiting to satisfy the condition.

Reply:
Changed it accordingly